File: cpp_10-99/12.cc
/**
 * C++
 * leetcode 12
 * url: https://leetcode-cn.com/problems/integer-to-roman/
 * 看了别人的，思路清晰明显不用解释
 */

class Solution {
public:
    string intToRoman(int num) {
        int values[]={1000,900,500,400,100,90,50,40,10,9,5,4,1};
        string reps[]={"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        
        string res;
        for(int i=0; i<13; i++){
            while(num>=values[i]){
                num -= values[i];
                res += reps[i];
            }
        }
        return res;
    }
};

File: Go/Go_10-99/12.go
// Go
// leetcode 12
// https://leetcode.cn/problems/integer-to-roman/

func intToRoman(num int) (res string) {
	values := []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
	strs := []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}
	for i := 0; i < 13; i++ {
		for num >= values[i] {
			num -= values[i]
			res += strs[i]
		}
	}
	return res
}